docs/library/uarray.rst: Add missing methods.

hlovatt opened the pull request.

hlovatt committed:
docs/library/uarray.rst: Add missing methods.

dpgeorge commented on docs/library/uarray.rst:
```diff
@@ -6,6 +6,11 @@
 |see_cpython_module| :mod:`python:array`.
 
+This module, ``uarray``, is a subset of ``python:array`` and only the methods documented below are available, 
+with the exception of ``decode``.
```
This sentence duplicates most of what is in the `see_cpython_module` replacement-macro just above.

dpgeorge commented on docs/library/uarray.rst:
```diff
@@ -27,3 +32,56 @@ Classes
+        The default for errors is 'strict', meaning that encoding errors raise a ``UnicodeError``. 
+        Other possible values are 'ignore' and 'replace'.
+        **Note:** ``decode`` is unique to MicroPython and behaves like ``array.tobytes().decode(...)`` does in Python
+        (but  is more efficient because it does not creat an intermediates ``bytes`` object).
```
I don't think `decode` was ever intended to exist on array instances.  It's there only for `bytearray` which is (for efficiency of code-size) implemented with the same code as other array type-codes.

So I'd suggest to never document this method on array because it shouldn't be used and may one day disappear.

hlovatt replied:
Would it be better to mark the method as deprecated? I suggest this because I discovered the method using `dir` and others might too.

dpgeorge commented:
Note that there are failures building the docs, see the "Build docs / build (pull_request)" link above.  Please try to correct these.

hlovatt committed:
docs/library/uarray.rst: Add missing methods.

Adds documentation for: decode, __getitem__, __setitem__, __len__, __add_, __iadd__, and __repr__.

hlovatt committed:
Update uarray.rst

hlovatt committed:
docs/library/uarray.rst: Add missing methods.

dpgeorge commented:
Merged in d2f8127258d91e4af5f5880e3519fc6d5cf879d9 with the following changes:
- removed docs for `decode` since it has been removed from the `array` class
- fixed docs styling
- fixed spelling and grammatical errors
